src/main/java/Server/MopeLSPServer.java

```diff
@@ -23,21 +29,57 @@
     private DiagnosticHandler diagnosticHandler;
     private static ICompilerAdapter compiler;
     private ConfigObject cfg;
+    private CompletableFuture<Object> shut;
+    private ServerSocket socket;
+    private String path;
 
     public MopeLSPServer(ConfigObject config){
         this.clients = new ArrayList<>();
         this.diagnosticHandler = new DiagnosticHandler(this);
-        this.compiler = new OMCAdapter("/usr/bin/omc", "us", "mope_local" );
+        readConfig();
+        this.compiler = new OMCAdapter(path, "us", "mope_local" );
         this.workspaceService = new MopeWorkspaceService(compiler);
         this.documentService = new MopeDocumentService(compiler);
         this.modelicaService = new MopeModelicaService(compiler, this);
         this.cfg = config;
+        this.shut = new CompletableFuture<>();
+    }
+
+    public void readConfigFile(String path) throws IOException {
```

Comment:
Why is this Function here?
The MopeServerLauncher already implemets this function and uses it to read the Config ...